azavea/pwd-fishway
# Fix font flicker/load times on staging

## Overview

Reduces the flickering/load times on staging by doing the following:

- **Ensure fonts are only requested once.** Due to a bug in Styled Components (styled-components/styled-components#1593), custom fonts were requested each time a component was mounted that referenced them. This resulted in the text on the site flickering when the fonts reloaded unnecessarily each time a new tab was clicked. One suggested workaround is to load the fonts through a CSS stylesheet, outside of the Styled Components pipeline. That workaround is implemented here.

- **Process images through webpack.** To reduce the number of resource requests, process the fish photos through webpack. This results in most of the images being Base64 encoded. These requests are faster and are cached. For some reason, with the current implementation, the images are requested each time the component is mounted. This does not happen when the images are processed through Webpack.

Connects #63

### Demo

See staging preview site.

### Notes

One of the main sources of the lag is the videos downloading. These requests block other requests because they are rather large. I've created a follow-up issue to try and lazy load the videos instead of loading them all at once on page load.

## Testing Instructions

- Compare the staging site to the build preview site for this PR, and verify the slowing and flickers that appear on staging are greatly reduced on the build preview branch.
